tests/sentry/rules/history/test_preview.py

```diff
@@ -4,64 +4,64 @@
 from freezegun import freeze_time
 
 from sentry.models import Activity, Group
-from sentry.rules.history.preview import PREVIEW_TIME_RANGE, get_hourly_bucket, preview
+from sentry.rules.history.preview import PREVIEW_TIME_RANGE, preview
 from sentry.testutils import TestCase
 from sentry.testutils.silo import region_silo_test
 from sentry.types.activity import ActivityType
 
 
+def get_hours(time: timedelta) -> int:
+    return time.days * 24 + time.seconds // (60 * 60)
+
+
 @freeze_time()
 @region_silo_test
 class ProjectRulePreviewTest(TestCase):
     def _set_up_first_seen(self):
-        hours = get_hourly_bucket(PREVIEW_TIME_RANGE)
+        hours = get_hours(PREVIEW_TIME_RANGE)
         for i in range(hours):
             for j in range(i % 5):
                 Group.objects.create(
                     project=self.project, first_seen=timezone.now() - timedelta(hours=i + 1)
                 )
-        return hours
 
     def _set_up_activity(self, condition_type):
-        hours = get_hourly_bucket(PREVIEW_TIME_RANGE)
+        hours = get_hours(PREVIEW_TIME_RANGE)
         for i in range(hours):
-            for j in range(i % 5):
-                Activity.objects.create(
-                    project=self.project,
-                    group=self.group,
-                    type=condition_type.value,
-                    datetime=timezone.now() - timedelta(hours=i + 1),
-                )
-        return hours
+            group = Group.objects.create(id=i, project=self.project)
+            Activity.objects.create(
+                project=self.project,
+                group=group,
+                type=condition_type.value,
+                datetime=timezone.now() - timedelta(hours=i + 1),
+            )
 
-    def _test_preview(self, hours, condition):
+    def _test_preview(self, condition, result1, result2):
         conditions = [{"id": condition}]
-        # test with 0 frequency, no fires should be filtered
         result = preview(self.project, conditions, [], "all", "all", 0)
-        for i in range(hours):
-            assert result[hours - i - 1].count == i % 5
+        assert len(result) == result1
 
-        # test with 60min frequency, there should be at most 1 fire per 60min bucket
-        result = preview(self.project, conditions, [], "all", "all", 60)
-        for i in range(hours):
-            assert result[i].count == (1 if i % 5 else 0)
+        result = preview(self.project, conditions, [], "all", "all", 120)
+        assert len(result) == result2
 
     def test_first_seen(self):
-        hours = self._set_up_first_seen()
+        self._set_up_first_seen()
         self._test_preview(
-            hours, "sentry.rules.conditions.first_seen_event.FirstSeenEventCondition"
+            "sentry.rules.conditions.first_seen_event.FirstSeenEventCondition", 670, 134
```

Comment:
can we make these some sort of static variable that's linked up with `i` and `j` somehow? bare integers can be hard to parse/understand

Reply:
tried to make some sense of it, but i think the numbers are pretty unintuitive to begin with

Reply:
no that's perfect. just like we expect them to move in proportion